### PR TITLE
Add toggle button at the end of the field

### LINKED ISSUE #51: It should be possible to add an toggle button/icon with a theme
A drop-down menu is commonly recognized by an arrow button at the end of the field.

It is possible to add it using a pseudo-element, but it’s really tricky and in some themes, the appropriate pseudo-elements are already in use for other purposes.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #51

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-dropdown-menu/60)

### CODE REVIEW

File: vaadin-dropdown-menu.html [hunk 1]
```diff
@@ -139,6 +158,14 @@
       :host([hidden]) {
         display: none !important;
       }
+
+      [part="toggle-button"] {
+        font-family: 'vaadin-dropdown-menu-icons';
```
Comment (jouni) [inline vaadin-dropdown-menu.html]: Would use double quotes for consitency

File: vaadin-dropdown-menu.html [hunk 2]
```diff
@@ -82,6 +90,15 @@
   </script>
 </dom-module>
 
+<style>
+  @font-face {
+    font-family: 'vaadin-dropdown-menu-icons';
```
Comment (jouni) [inline vaadin-dropdown-menu.html]: Double quotes here as well.

File: vaadin-dropdown-menu.html (hunk 1 above)
Comment (web-padawan) [reply vaadin-dropdown-menu.html]: Done.

File: vaadin-dropdown-menu.html (hunk 2 above)
Comment (web-padawan) [reply vaadin-dropdown-menu.html]: Done.

**Comment by tomivirkki** [comment]
:lgtm:

---

Reviewed 6 of 8 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-:-Kz3zGhIea2Uq0-eueNC:bnfp4nl)*

**Comment by web-padawan** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 95 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz0-1jNID4FoeIsxcK8-r1-95:-Kz5_W09t46MV3Z6WHT2:b-896fix) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/3a6c39d00aa9e32cc232e7ce5bd81568a773ca24/vaadin-dropdown-menu.html#L95)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Double quotes here as well.
</blockquote></details>

Done.

---

*[vaadin-dropdown-menu.html, line 163 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz0-1jNID4FoeIsxcK8-r1-163:-Kz5_WptsA__M131t4Dt:b-896fix) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/3a6c39d00aa9e32cc232e7ce5bd81568a773ca24/vaadin-dropdown-menu.html#L163)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Would use double quotes for consistency.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by manolo** [comment]
Needs rebasing with master, and probably updating screenshots

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz7c0sKZi4gmIK9xHsh:bcalk3b) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
> ```HTML
>           return [
>             '_valueChanged(value, _items)',
>             '_updateAriaExpanded(opened)'
> ```


Instead of this,  move the opened observer here

`'_openedChanged(opened, _toggleElement)'` 

and move the content of the `_updateAriaExpanded(opened)` to the `_openedChanged` block

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-:-Kz7cRHj40jCJFpI37PW:b-5g9y4y)*

**Comment by manolo** [comment]
I did a local rebase with master, and screenshots look wrong, for some browsers the selected text is broken in two lines and in IE11 the toggle button is not aligned to the right but at the end of the selected text. Take a look to these screenshots:




---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-:-Kz7h_zqA00JsP0Kp3rh:b-66stzj)*

**Comment by web-padawan** [comment]
Review status: 1 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz8GbCGHyKUHehWC1ye:b-iwadps) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
> _toggleElement
That would require refactoring code relying on `wasOpened`. So I added another check to the `_openedChanged` and moved the `_updateAriaExpanded ` call there as you suggested.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by tomivirkki** [comment]
Reviewed 7 of 7 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by web-padawan** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz8vrhm3fh5VJyXiZeA:b-rb0tiw) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

> _toggleElement
That would require refactoring code relying on `wasOpened`. So I added another check to the `_openedChanged` and moved the `_updateAriaExpanded ` call there as you suggested.
</blockquote></details>

My comment above looks ugly, so I'll describe it once again in more detail:
- existing observer `_openedChanged ` uses `wasOpened` check, which won't be an option in the multi-property observer;
- since observers in Polymer 2 run first time before `ready`, we still have to run `_updateAriaExpanded ` there as well.

Please point me out if I'm missing something and there is a better solution so far.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz9-p7UjoGKv6zn67w_:b-btt0uk) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

My comment above looks ugly, so I'll describe it once again in more detail:
- existing observer `_openedChanged ` uses `wasOpened` check, which won't be an option in the multi-property observer;
- since observers in Polymer 2 run first time before `ready`, we still have to run `_updateAriaExpanded ` there as well.

Please point me out if I'm missing something and there is a better solution so far.
</blockquote></details>

I see. Then the code should be this, and you don't need to call _updateAriaExpanded in the ready block

```

_updateAriaExpanded(opened, _toggleElement)

...

_updateAriaExpanded(opened, toggleElement) {
  if (toggleElement) {
    toggleElement.setAttribute('aria-expanded', opened);
  }
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz90EWcuFid8nJmvB9-:b-fts9ut) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I see. Then the code should be this, and you don't need to call _updateAriaExpanded in the ready block

```

_updateAriaExpanded(opened, _toggleElement)

...

_updateAriaExpanded(opened, toggleElement) {
  if (toggleElement) {
    toggleElement.setAttribute('aria-expanded', opened);
  }
}
```

</blockquote></details>

Simpler

```
_updateAriaExpanded(opened, toggleElement) {
  toggleElement && toggleElement.setAttribute('aria-expanded', opened);
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by web-padawan** [comment]
Review status: 6 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 343 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-Kz7c0sKZi4gmIK9xHsg:-Kz94vr63XzbquLsBrJv:b-epmf15) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/856abb19fc0e5a4fd5b6e6e4aebd663af05e27e2/vaadin-dropdown-menu.html#L343)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Simpler

```
_updateAriaExpanded(opened, toggleElement) {
  toggleElement && toggleElement.setAttribute('aria-expanded', opened);
}
```
</blockquote></details>

Done. I should have solved it myself :) Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by manolo** [comment]
Reviewed 1 of 8 files at r1, 5 of 7 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Comment by manolo** [comment]
:lgtm:  good Job !

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60#-:-Kz983xHRSw2Re93eHJP:bmymdn0)*

**Comment by manolo** [comment]
Reviewed 1 of 8 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/60)*

**Pull request merged**